src/PhpSpreadsheet/Shared/StringHelper.php

```diff
@@ -560,6 +562,22 @@ public static function convertToNumberIfFraction(string &$operand): bool
 
     //    function convertToNumberIfFraction()
 
+    /**
+     * Identify whether a string contains a percentage, and if so,
+     * convert it to a numeric.
+     *
+     * @param string $operand string value to test
+     */
+    public static function convertToNumberIfPercent(string &$operand): bool
+    {
+        if (preg_match(self::STRING_REGEXP_PERCENT, $operand, $match)) {
```

Comment:
There's a couple of little tricks that we can do with the regexp to extract the numeric value directly, rather than doing the `str_replace()` call.
```
~^(?: *-? *\% *(?<PrefixedValue>[0-9]+\.?[0-9*]*) *| *\-? *(?<PostfixedValue>[0-9]+\.?[0-9]*) *\% *)$~
```
To avoid capture of the complete string, I've set that as a non-capture group, because we're only interested in whether the `%` character appears, we don't need to actually capture it or any padding around it; and then I've set capture groups for the actual numeric part of the string. I've also named these groups (`PrefixedValue` and `PostfixedValue`) so that they can be referenced by name in the `$matches` array.

Then, if a match is identified, we can extract the numeric part directly by name, whether it's prefixed by the `%` or postfixed.
```php
$poperand = (float) ($match['PostfixedValue'] ?? $match['PrefixedValue']) / 100.0;
```
The null coalescence identifies whether the number is prefixed or postfixed (using the named groups from `$matches`); and then does the division by 100 and casts the result to a float. There's no function call overhead; just PHP operations, so it should be more efficient if it's being called in a loop thatr checks a lot of cell values.

Reply:
Great trick with the regexp.  I agree that eliminating the `str_replace` would be a big improvement.
Unfortunately the changes to the regexp break the negative percentage cases.  I'll see if I can run with your idea and get it working for negative values.

Reply:
Ah, yes! I was being a bit to quick and casual with the regexp changes. If the capture group is expanded to encompass the sign as well, then we're potentially also capturing whitespace too, which would break other things.

One option would be to have additional pre/postfix capture groups for the sign, and a match there would trigger a `* -1` operation
```
~^(?: *(?<PrefixedNegative>-?) *\% *(?<PrefixedValue>[0-9]+\.?[0-9*]*) *| *(?<PrefixedNegative>\-?) *(?<PostfixedValue>[0-9]+\.?[0-9]*) *\% *)$~
```
and then
```php
$sign = $match['PostfixedNegative'] ?? $match['PrefixedNegative'];
$poperand *= ($sign === '-') ? -1.0 : 1.0;
```
So we're still using only operations, and no function calls.

Just to be bloody minded and awkward, MS Excel recognises a `%` prefix with the sign before or after the `%`, so `%-2.5` and `-%2.5` are both recognised as numeric values. Fortunately with a postfix `%`, the sign must be before the numeric part.
And, of course, the sign could be a `+` as well as a `-`; and the numeric part could be in scientific format, `-2.5E-4%`.

But if you get the basic negative working, I'll merge and then do some more tweaking myself to handle a few of those edge cases; and refactor it all into a separate helper.

I did say right from the start that this was a lot more complicated once you start to look at it in detail. ;-)

Reply:
You did warn me! I can't imagine what the code for Excel's formula parser must look like.
I'll take a crack at improving the regexp like you suggest.  

Should we be concerned about any of the other spreadsheet applications?  I'm focused entirely on Excel because it was the root of my issue, but I suppose it's worth considering how Open Office or Libre Office works?  Won't that be a nightmare if they handle these various combinations of `%`, `+`, and `-` differently....  I don't have either installed at the moment but I'll download a copy tonight and see how they behave.

Reply:
Okay sounds good.  I was able to get it working for combinations of `+`, `-`, and scientific notation by expanding on the approach you suggested.  
I broke out all the different pieces of the final calculation because it was getting a bit difficult to follow as single line, owing to all the null coalescing checks.  We could simplify it by moving all those checks inline.  
Note that, as I mention in the commit, I did have to add the `PREG_UNMATCHED_AS_NULL` flag to the `preg_match`, otherwise I was getting '' back when a capture group wasn't found.
Let me know what you think.

Reply:
Updated my Excel sheet with various permutations.
[Excel Test Cases.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/9963751/Excel.Test.Cases.xlsx)